File: src/fe/src/models/MaterialClinicalDetail.ts
import { Gender } from "@/models/enums/Gender";
import { ShipmentMaterialCondition } from "@/models/enums/ShipmentMaterialCondition";

export interface MaterialClinicalDetail {
  Id: string;
  MaterialShippingInformationId: string;
  MaterialProductId: string;
  TransportCategoryId: string;
  MaterialNumber: string;
  CollectionDate: Date;
  Location: string;
  IsolationHostTypeId: string;
  Gender: Gender;
  Age: number;
  PatientStatus: string;
  Condition: ShipmentMaterialCondition;
  Note: string;
}

File: src/be/src/WHO.BioHub.Models/Models/MaterialClinicalDetail.cs
using WHO.BioHub.Models.Models.Abstractions;
using WHO.BioHub.Shared.Enums;

namespace WHO.BioHub.Models.Models;

public class MaterialClinicalDetail : EntityBase
{
    public Guid? MaterialShippingInformationId { get; set; }
    public virtual MaterialShippingInformation MaterialShippingInformation { get; set; }
    public string MaterialNumber { get; set; }
    public DateTime? CollectionDate { get; set; }
    public string Location { get; set; }
    public Guid? IsolationHostTypeId { get; set; }
    public virtual IsolationHostType IsolationHostType { get; set; }
    public Gender? Gender { get; set; }
    public int? Age { get; set; }
    public string PatientStatus { get; set; }
    public ShipmentMaterialCondition? Condition { get; set; }
    public string Note { get; set; }
}
